src/Pages/Detail/DetailMain.js

```diff
@@ -1,115 +1,119 @@
-        { title: "오픈 도어", bgc: "#D05A57" },
-        { title: "풀 크레딧", bgc: "#BA0C2F" },
-      ],
-    };
-  }
-
-  plusOne = () => {
-    if (this.state.count >= 5) {
-      alert("최대 주문수량은 5개 입니다.");
-      return;
+    constructor(props) {
+        super(props);
+        this.state ={
+            count: 1,
+            totalPrice: null,
+            colorList :[]
+        }
     }
-    this.setState({
-      count: this.state.count + 1,
-      totalPrice: this.state.totalPrice + 19000,
-    });
-  };
 
-  minusOne = () => {
-    if (this.state.count <= 1) {
-      alert("최소 주문수량은 1개 입니다");
-      return;
+
+    componentDidMount =() => {
+        console.log(this.state.colorList)
+        fetch('http://localhost:3000/data/data.json')
+        .then(res => res.json())
+        .then(res => {
+            this.setState({
+                colorList: res.colorList,
+            })
+        })
+    }
+    
+    plusOne = () => {
+        if(this.state.count >= 5){
+            alert('최대 주문수량은 5개 입니다.')
+            return;
+        }
+        this.setState({
+            count: this.state.count + 1,
+            totalPrice: this.props.price * (this.state.count + 1)
+        });   
+        // console.log("hi",this.state.totalPrice)
+         
+    }
+    
+    minusOne = () => {
+        const{ count } = this.state
+        const{ price } = this.props
+        if(count <= 1){
+            alert('최소 주문수량은 1개 입니다')
+            return;
+        }
+        this.setState({
+            count: count - 1,
+            totalPrice: price * (count - 1)
+        })
     }
-    this.setState({
-      count: this.state.count - 1,
-      totalPrice: this.state.totalPrice - 19000,
-    });
-  };
+     
 
-  render() {
-    return (
-      <div className="DetailMain">
-        <div className="mainContainer">
-          <div className="detail_mainImg">
-            <img src="Images/detail-main.jpg" alt="main Image" />
-          </div>
-          <div className="productInfo">
-            <div className="productName">
-              <span>무드인핸서 리퀴드 마뜨 얼루어먼트</span>
+    render() {
+        const{name, mainImage, price} = this.props;
+         
+        return (
+            <div className="DetailMain">
+                <div className="mainContainer">
+                    <div className="detail_mainImg">
+                        <img src= {mainImage}  alt="main-image" />
+                    </div>
+                    <div className="productInfo">
+                        <div className="productName">
+                        <span>{name}</span>
+                        </div>
+                        <div className="productElement">
+                        <div className="productPrice">
+                            <div className="nameBox">
+                            <span className="detailName">판매가</span>
+                            </div>
+                        <span className="price">{parseInt(price).toLocaleString()}원</span>
```

Comment:
```suggestion
                        <span className="price">{+price.toLocaleString()}원</span>
```
parseInt는 성능 이슈가 있어서 +나 Number() 사용 고려해주세요!